### PR TITLE
Contributing lib/ansible/modules/network/cloudengine/ce_snmp_traps.py module to manage HUAWEI data center CloudEngine

### PR DESCRIPTION
**Pull request by QijunPan**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ce_snmp_traps.py
##### ANSIBLE VERSION

```
ansible 2.3.0
config file = /etc/ansible/ansible.cfg
configured module search path = ['/home/ansible-2.1.1.0/lib/ansible/modules/core/network/cloudengine', '/home/ansible-2.1.1.0/lib/ansible/module_utils', '/home/ansible-2.1.1.0/lib/ansible/modules', '/home/ansible-2.1.1.0', '/home/ansible-2.1.1.0/lib/ansible/utils/']
```

##### SUMMARY

```
TASK [config snmp traps] *******************************************************
task path: /usr1/code/OpenNESS/code/current/KVM/intg/test/testcases/test-ce_snmp_traps.yml:18
Using module file /usr/local/lib/python2.7/site-packages/ansible-2.3.0-py2.7.egg/ansible/modules/core/network/cloudengine/ce_snmp_traps.py
<ce_6850> ESTABLISH LOCAL CONNECTION FOR USER: root
<ce_6850> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693 `" && echo ansible-tmp-1487053737.44-173263156287693="` echo $HOME/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693 `" ) && sleep 0'
<ce_6850> PUT /tmp/tmpzUPYUK TO /root/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693/ce_snmp_traps.py
<ce_6850> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693/ /root/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693/ce_snmp_traps.py && sleep 0'
<ce_6850> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693/ce_snmp_traps.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1487053737.44-173263156287693/" > /dev/null 2>&1 && sleep 0'
changed: [ce_6850] => {
    "changed": true, 
    "end_state": {
        "snmp-agent trap": [
            "enable"
        ], 
        "undo snmp-agent trap": []
    }, 
    "existing": {
        "snmp-agent trap": [], 
        "undo snmp-agent trap": []
    }, 
    "invocation": {
        "module_args": {
            "auth_pass": null, 
            "authorize": false, 
            "feature_name": "all", 
            "host": "ce_6850", 
            "interface_number": null, 
            "interface_type": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 12345, 
            "port_number": null, 
            "provider": null, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": null, 
            "trap_name": null, 
            "use_ssl": false, 
            "username": "rootDC", 
            "validate_certs": true
        }, 
        "module_name": "ce_snmp_traps"
    }, 
    "proposed": {
        "feature_name": "all", 
        "state": "present"
    }, 
    "updates": [
        "snmp-agent trap enable"
    ]
}
```

### CODE REVIEW

File: lib/ansible/modules/network/cloudengine/ce_snmp_traps.py [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+                                   'mpls_lspm', 'msdp', 'mstp', 'nd', 'netconf', 'nqa', 'nvo3',
+                                   'openflow', 'ospf', 'ospfv3', 'pim', 'pim-std', 'qos', 'radius',
+                                   'rm', 'rmon', 'securitytrap', 'smlktrap', 'snmp', 'ssh', 'stackmng',
+                                   'sysclock', 'sysom', 'system', 'tcp', 'telnet', 'trill', 'trunk',
+                                   'tty', 'vbst', 'vfs', 'virtual-perception', 'vrrp', 'vstm', 'all']),
+        trap_name=dict(type='str'),
+        interface_type=dict(choices=['Ethernet', 'Eth-Trunk', 'Tunnel', 'NULL', 'LoopBack', 'Vlanif',
+                                     '100GE', '40GE', 'MTunnel', '10GE', 'GE', 'MEth', 'Vbdif', 'Nve']),
+        interface_number=dict(type='str'),
+        port_number=dict(type='str')
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_traps.py:539]: For integers please:
use `type='int`
remove the `isdigit` check from `check_params

File: lib/ansible/modules/network/cloudengine/ce_snmp_traps.py [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+        if self.port_number:
+            if self.port_number.isdigit():
+                if int(self.port_number) < 1025 or int(self.port_number) > 65535:
+                    self.module.fail_json(
+                        msg='Error: The value of port_number is out of [1025 - 65535].')
+            else:
+                self.module.fail_json(
+                    msg='Error: The port_number is not digit.')
+
+        if self.interface_type:
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_traps.py]: There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```

File: lib/ansible/modules/network/cloudengine/ce_snmp_traps.py [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+            if not self.interface_number:
+                self.module.fail_json(
+                    msg='Error: The interface_type and interface_number should input at the same time.')
+        if self.interface_number:
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_traps.py]: There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/cloudengine/ce_snmp_traps.py (hunk 2 above)
Comment (QijunPan) [reply lib/ansible/modules/network/cloudengine/ce_snmp_traps.py]: I'll fix it. thank you.

File: lib/ansible/modules/network/cloudengine/ce_snmp_traps.py (hunk 3 above)
Comment (QijunPan) [reply lib/ansible/modules/network/cloudengine/ce_snmp_traps.py]: I'll fix it. thank you.

File: lib/ansible/modules/network/cloudengine/ce_snmp_traps.py (hunk 1 above)
Comment (QijunPan) [reply lib/ansible/modules/network/cloudengine/ce_snmp_traps.py:539]: Because the device side is string, avoid repeated conversion,  retain it， thanks.

**Comment by QijunPan** [comment]
@gundalow   These problems have been fixed， can  this PR merged?

**Review by gundalow — APPROVED** [review]

**Pull request merged**